source/Tutorials/Advanced/Security/Deployment-Guidelines.rst

```diff
@@ -0,0 +1,214 @@
+
+A basic core guidelines on how to organize the different certificates, keys and directories remains to be a critical factor in order to avoid compromising the security of the system.
+This includes protection-awareness and criteria for selecting the minimum set of necessary files to be deployed upon remote production systems for minimizing security exposure.
+
+Prerequisites
+-------------
+
+* A docker installation available. Please refer to the installation steps detailed in `Docker installation <https://docs.docker.com/engine/install/>`_.
+* (Recommended) A basic understanding on `ROS2 Security design <https://design.ros2.org/articles/ros2_dds_security.html>`_.
+* (Recommended) Previous tutorials completion.
+
+General Guidelines
+------------------
+
+ROS2 leverages DDS Security extension to ensure security on message exchanges within the same enclave.
+The different signed files and certificates within an enclave are generated from the private keys and certificates of a `Certificate Authority (CA) <https://en.wikipedia.org/wiki/Certificate_authority>`_ trusted entity.
+In fact, two different CA's can be selected for identity and permissions, per enclave.
+Those CA artifacts are stored inside ``private/`` and ``public/`` sub-directories of a `Keystore <https://design.ros2.org/articles/ros2_security_enclaves.html>`_ with the following folder structure:
+
+.. code-block:: text
+
+  keystore
+  ├── enclaves
+  │   └── ...
+  │       └── ...
+  ├── private
+  │   └── ...
+  └── public
+      └── ...
+
+A good practice for the creation and usage of a certain Certificate Authority on a typical deployment over a production system, is to:
+
+  (1) Create it within the organization system intended for internal use only.
+  (2) Generate/modify desired enclaves.
+  (3) Ship ``public/`` alongside with corresponding ``enclaves/`` into the different remote production systems during setup, depending on the communication needs (not all remote systems may need to interact with each other).
+  (4) Keep and protect ``private/`` keys and/or certification requests in the organization.
+
+It is important to note that if ``private/`` files are lost, it won't be possible to change access permissions, adding or modifying security profiles anymore.
+
+In addition, further practices may be taken into consideration:
+
+* Granting read-only permissions to the ``enclaves/`` directory contents.
+* If a PKCS#11 compliant URI is given for generating enclave's private keys, a `Hardware Security Module (HSM) <https://en.wikipedia.org/wiki/Hardware_security_module>`_ could be used to store them.
+
+The following table depicts a summary of the previous statements relating the Keystore directory with the Recommended location:
+
++------------------------+--------------+---------------+---------------------+
+| Directory / Location   | Organization | Remote System | Material Sensitivity|
```

Comment:
In that way I think we could go for "Target Device"

Reply:
I kept the `remote_system` name just for the tutorial section (otherwise i would need to re-do the video). In the first paragraph of the tutorial, it is specified to be `...remote production target device actor` in reference to the table above.